### PR TITLE
results.openaddress.io support

### PR DESCRIPTION
**Pull request by ingalls**
- Add support for URLs with dots in the bucket name & assoc. tests
    - https://github.com/mapbox/s3urls/pull/12/files#diff-168726dbe96b3ce427e7fedce31bb0bcR26
- cleans up project to use node6 w/ geocoding eslint

Closes: https://github.com/mapbox/s3urls/issues/10
Closes: https://github.com/mapbox/s3urls/pull/1

Refs: https://github.com/mapbox/mapbox-tile-copy/issues/53

### CODE REVIEW

**Review by yhahn — DISMISSED** [review]
@ingalls caan you leave indenting alone and use http://github.com/mapbox/eslint-config-mapbox/ instead for linting (I believe it uses 2 spaces)

Will make it much easier for me to read & review.

File: index.js [hunk 1]
```diff
@@ -21,7 +23,12 @@ var s3urls = module.exports = {
       key = uri.pathname.split('/').slice(2).join('/');
     }
     if (style === 'bucket-in-host') {
-      bucket = uri.hostname.split('.')[0];
+      const match = uri.hostname.match(new RegExp(/.*?(?=.s3)/));
+      if (match) {
+        bucket = match[0];
+      } else {
+        bucket =  uri.hostname.split('.')[0];
+      }
```
Comment (yhahn) [inline index.js]: @ingalls am I right that the above is the only substantive change in the library (plus your additional tests) ?

Comment (ingalls) [reply index.js]: Yes

Comment (yhahn) [reply index.js]: Can you add a testcase for `results.s3llout-to-the-man.io.s3.amazonaws.com`

Comment (ingalls) [reply index.js]: good :eyes: , done

**Review by yhahn — APPROVED** [review]
:+1: Looks great, and thanks for sprucing up the node version testing & linting situation. Looking forward to the next uncovered cornercase!!

**Pull request merged**